README.md

```diff
@@ -4,130 +4,180 @@
+
+ 1. **Advanced ML Researchers** who are limited by current ML frameworks.
+    S4TF's advantages include a seamless integration with a modern general-purpose
+    language, allowing for more dynamic and sophisticated models. Fast
+    abstractions can be developed "in user-space" (as opposed to in C/C++
+    aka "framework-space"), resulting in modular APIs that can be easily
+    customized.
+
+ 2. **ML Learners** who are just getting started with machine learning. Thanks
+    to Swift's support for quality tooling (e.g. context-aware autocomplete),
+    S4ML can be one of the most productive ways to get started learning the
+    fundamentals of machine learning.
+
+## Getting started
+
+ - **Colab**: The fastest way to get started is to try out S4TF right in your
+   browser. Just open up our [getting started
+   notebook](https://colab.research.google.com/github/tensorflow/swift-tutorials/blob/master/iris/swift_tensorflow_tutorial.ipynb) (or start from a
+   [blank notebook](https://colab.research.google.com/github/tensorflow/swift/blob/master/notebooks/blank_swift.ipynb))!
+   Read more in our [usage guide](Usage.md).
+
+ - **Install locally**: you can [download a pre-built Swift for TensorFlow
+   package](Installation.md). After installation, you can follow these
+   [step-by-step instructions](Usage.md) to build and execute a S4TF program
+   on your computer.
+
+ - **Compile from source**: If you'd like to customize S4TF or even contribute
+   back, follow our [instructions on building the S4TF compiler from source,
+   from in the compiler's README](https://github.com/apple/swift/tree/tensorflow).
+
+Please do join the
+[swift@tensorflow.org mailing list](https://groups.google.com/a/tensorflow.org/d/forum/swift)
+to hear the latest announcements, get help, and share your thoughts!
+
+## Why S4TF?
 
 Swift for TensorFlow is a new way to develop machine learning models. It
 gives you the power of
 [TensorFlow](https://www.tensorflow.org) directly integrated into the
-[Swift programming language](https://swift.org/about).
-With Swift, you can write the following imperative code, and Swift 
-automatically turns it into **a single TensorFlow Graph** and runs it 
-with the full performance of TensorFlow Sessions on CPU, GPU and 
-[TPU](https://cloud.google.com/tpu).
+[Swift programming language](https://swift.org/about). We believe that
+machine learning paradigms are so important that they deserve
+**first-class language and compiler support**. 
+
+Optimizing parameters by taking the gradient of functions is a fundamental
+primitive of machine learning. With Swift for TensorFlow,
+you can write a normal function using arbitrary data types, and then
+take the derivative of it with just a single annotation.
 
 ```swift
 import TensorFlow
 
 var x = Tensor<Float>([[1, 2], [3, 4]])
 
-for _ in 1...5 {
-    x += x • x // alternatively: x += matmul(x, x)
-    print(x)
+@differentiable
+func myWackyPowPlus(ofBase base: Tensor<Float>,
```

Comment:
I think a different example may be more suitable here.

`myWackyPowPlus` is actually not differentiable wrt the `exponent: Int` argument:
`error: can only differentiate with respect to parameters that conform to 'Differentiable', but 'Int' does not conform to 'Differentiable'`

Also, I think including an example call to a differential operator like `gradient(of:)` at the bottom of the code snippet is crucial.

---

I'll update this comment with an updated AD example in a bit!

Reply:
I don't really like showing `@differentiable` because it doesn't show any actual differentiation. Showing the use of a differential operator (e.g. `Differentiable.gradient(in:)`) with a trailing closure is most clear.

Reply:
Here are two examples demonstrating differential operators:
```swift
import TensorFlow
let 𝛁x = gradient(at: 4.0) { x in
  x * x * x
}
print(𝛁x) // 48.0
```

```swift
// Custom differentiable type.
struct Model : Differentiable {
  var w: Float
  var b: Float
  func applied(to input: Float) -> Float {
    return w * input + b
  }
}
// Differentiate using `Differentiable.gradient(at:)`.
let (𝛁model, 𝛁input) = Model(w: 4.0, b: 3.0).gradient(at: 2.0) { (model, input) in
  model.applied(to: input)
}
print(𝛁model) // Model.AllDifferentiableVariables(w: 2.0, b: 1.0)
print(𝛁input) // 4.0
```

I'd recommend using the second one on the README.

Reply:
I like the second example, but the closure sticks out on the right hand side. Moving `Model(w: 4.0, b: 3.0)` to a temporary variable would be better.

Reply:
I've updated to the second one, as that's pretty nice. But what do you think about making it use Tensors instead of just floats? Or a mix of Tensors and standard floats?

Reply:
+1 to mixing.

Reply:
Except you'll end up with a super wacky model.